pkgs/development/libraries/slang/terminfo-dirs.patch

```diff
@@ -0,0 +1,43 @@
+diff --git a/src/sltermin.c b/src/sltermin.c
+index a06d0e4..200e503 100644
+--- a/src/sltermin.c
++++ b/src/sltermin.c
+@@ -276,14 +276,7 @@ static SLCONST char *Terminfo_Dirs [] =
+ #ifdef MISC_TERMINFO_DIRS
+    MISC_TERMINFO_DIRS,
+ #endif
+-   "/usr/local/etc/terminfo",
+-   "/usr/local/share/terminfo",
+-   "/usr/local/lib/terminfo",
+    "/etc/terminfo",
+-   "/usr/share/terminfo",
+-   "/usr/lib/terminfo",
+-   "/usr/share/lib/terminfo",
+-   "/lib/terminfo",
+    NULL,
+ };
+ 
+@@ -352,6 +345,23 @@ SLterminfo_Type *_pSLtt_tigetent (SLCONST char *term)
+ 	Terminfo_Dirs [1] = home_ti;
+      }
+ 
++   if (NULL != (env = _pSLsecure_getenv ("TERMINFO_DIRS"))) {
++     void * tidirs_env = strdup(env);
++     tidir = strtok(tidirs_env, ":");
++     do {
++       if (sizeof (file) > strlen (tidir) + 5 + strlen (term))
```

Comment:
Did you mean strlen instead of sizeof  ? `sizeof file` would be the size of a pointer if I understand C correctly.

Reply:
In the original source, `file` is declared as `char file[1024];`. `sizeof file` is thus the amount of space available at file, so this is correct — the condition is "do we have enough space at `file` to store the full path to the terminfo file". C sucks. :p

Reply:
Ah sorry, I should have read the whole file. Anyway using strlen here would not have made sense.